ATParser.cpp

```diff
@@ -155,7 +155,8 @@ int ATParser::vscanf(const char *format, va_list args)
         sscanf(_buffer+offset, _buffer, &count);
 
         // We only succeed if all characters in the response are matched
-        if (count == j) {
+        // and we are at the end of the response
+        if ((count == j) && (format[i-1] == (const char)c)) {
```

Comment:
It's been a while since I've looked at this, so I may be a bit confused, but shouldn't `count == j` mean that `format[i-1] == (const char)c` is always true? When is the second check needed?

Is scanf not working?

@sarahmarshy, can you review this and see if this might cause issues with other drivers? If it shouldn't we can probably merge it in and test drivers when they update the ATParser library.